src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -478,63 +487,104 @@ internal static IEnumerable<TrainTestData> CrossValidationSplit(IHostEnvironment
 
                 var testFilter = new RangeFilter(env, new RangeFilter.Options
                 {
-                    Column = samplingKeyColumnName,
+                    Column = splitColumn,
                     Min = (double)fold / numberOfFolds,
                     Max = (double)(fold + 1) / numberOfFolds,
                     Complement = false,
                     IncludeMin = true,
                     IncludeMax = true
                 }, data);
 
-                yield return new TrainTestData(trainFilter, testFilter);
+                var trainDV = ColumnSelectingTransformer.CreateDrop(env, trainFilter, splitColumn);
```

Comment:
>CreateDrop [](start = 57, length = 10)

Is this a new behavior? Or is it needed now because of the other changes?
If it is a new behavior, why is it needed?

Reply:
Dropping the SplitColumn is something new to fix the situation I mentioned here:
[https://github.com/dotnet/machinelearning/pull/5227#discussion_r446491522](https://github.com/dotnet/machinelearning/pull/5227#discussion_r446491522)

Not dropping it was causing an issue with AutoML. And also it didn't make sense that we didn't drop it, because if we didn't do it then the schema of a DataView was changed by splitting it, which I think should be considered unexpected behavior.

---
In reply to: [447415004](https://github.com/dotnet/machinelearning/pull/5227#discussion_r447415004) [](ancestors = 447415004)